src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/BotSegmentZipParser.cs

```diff
@@ -30,24 +85,61 @@ public static List<BotSegment> ParseBotSegmentZipFromSystemPath(string zipFilePa
                 foreach (var entry in entries)
                 {
                     using var sr = new StreamReader(entry.Open());
-                    var frameData = JsonConvert.DeserializeObject<BotSegment>(sr.ReadToEnd());
 
-                    if (frameData.EffectiveApiVersion > SdkApiVersion.CURRENT_VERSION)
+                    var fileText = sr.ReadToEnd();
+
+                    try
                     {
-                        versionMismatch = frameData.EffectiveApiVersion;
-                        break;
-                    }
+                        // see if this is a bot segment list file
+                        var botSegmentList = JsonConvert.DeserializeObject<BotSegmentList>(fileText);
 
-                    badSegmentName = entry.FullName;
+                        if (botSegmentList.EffectiveApiVersion > SdkApiVersion.CURRENT_VERSION)
+                        {
+                            versionMismatch = botSegmentList.EffectiveApiVersion;
```

Comment:
Just want to make sure I understand this, so this now allows the bot segment list to set the version for the underlying segments so that each segment doesn't need a version, is that a correct statement?

Reply:
The effective api version checking system HAS NOT CHANGED.

What I did is removed the apiVersion as a required field on every element.

When saving bot segments during recording they will still be written there and enforced.

When using our UI tools, they will also still be written and enforced.

What this change allows is for hand written files to not need to specify them and they will default to being the 'latest' version for each model present.  Makes developers editing these files way less tedious.  What I did as a 'hybrid' was to put the version I wanted just at the top level of each of my files, which made it much easier to manage/understand when hand editing the files.

(If you were to load one of these files on an older SDK version, it may or may not load depending on which fields are there and you just wouldn't get the clean 'version unsupported' message.)

Reply:
I could see an argument for still making it required on BotSegmentLists and maybe BotSegments at the top level

Reply:
Got it, that seems fine to me!